mono/metadata/class-init.c

```diff
@@ -4128,7 +4122,8 @@ mono_class_get_object_finalize_slot ()
 MonoMethod *
 mono_class_get_default_finalize_method ()
 {
-	return default_finalize;
+	int const i = finalize_slot;
+	return (i < 0) ? NULL : mono_defaults.object_class->vtable [i];
 }
```

Comment:
Is this related to the write barrier changes?

Reply:
This PR is a smaller form, yes.
And this code initializes two globals, with one guard.
My change is to initialize one global, with one guard.